core/src/main/java/com/orientechnologies/orient/core/storage/ridbag/sbtree/OIndexRIDContainer.java

```diff
@@ -55,8 +56,13 @@
   /**
    * Should be called inside of lock to ensure uniqueness of entity on disk !!!
    */
-  public OIndexRIDContainer(String name, boolean durableNonTxMode) {
-    fileId = resolveFileIdByName(name + INDEX_FILE_EXTENSION);
+  public OIndexRIDContainer(String name, boolean durableNonTxMode, AtomicLong bonsayFileId) {
+    if (bonsayFileId.get() == 0) {
+      fileId = resolveFileIdByName(name + INDEX_FILE_EXTENSION);
+      bonsayFileId.set(fileId);
+    } else {
+      fileId = bonsayFileId.get();
```

Comment:
I suppose to avoid double get of volatile field which will be cached in  most of the cases better to do 
```
fileId = bonsayFileId.get();
if (fileId == 0) {
      fileId = resolveFileIdByName(name + INDEX_FILE_EXTENSION);
   bonsayFileId.set(fileId);
} 
```

that is minor so up to you